src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1718,16 +1873,70 @@ WorkerAggregateWalker(Node *node, List **newExpressionList)
 /*
  * WorkerAggregateExpressionList takes in the original aggregate function, and
  * determines the transformed aggregate functions to execute on worker nodes.
- * The function then returns these aggregates in a list.
+ * The function then returns these aggregates in a list. It also creates
```

Comment:
creating group by references is moved to ```WorkerExtendedOpNode()```. Now we generate a group by clause for each Var node returned from ```WorkerAggreagateWalker()```.